### PR TITLE
Mono transition

### PR DESCRIPTION
**Pull request by maschmann**
## Description

Move the manyrepos to one monorepo for better maintainability

### CODE REVIEW

File: src/PhpFlo/Fbp/Tests/FbpDefinitionTest.php [hunk 1]
```diff
@@ -0,0 +1,130 @@
+<?php
```
Comment (aretecode) [inline src/PhpFlo/Fbp/Tests/FbpDefinitionTest.php]: weird, is it just for me that this file isn't highlighted in the review page? https://github.com/phpflo/phpflo/pull/38/files#diff-2cc6896b59caf9159e948b1004a5c344R3

**Review by aretecode — APPROVED** [review]
👍

File: src/PhpFlo/Common/DefinitionInterface.php [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+    /**
+     * @return array
+     */
+    public function toArray() : array;
+
+    /**
+     * @return string
+     */
+    public function toJson() : string;
```
Comment (maschmann) [inline src/PhpFlo/Common/DefinitionInterface.php]: nothing official,yet - but it seems to condense around ():<space><type>
Adjusted accordingly :-)

File: src/PhpFlo/Common/HookableNetworkInterface.php [hunk 3]
```diff
@@ -10,8 +10,8 @@
 declare(strict_types=1);
```
Comment (maschmann) [inline src/PhpFlo/Common/HookableNetworkInterface.php]: adjusted accordingly

File: src/PhpFlo/Common/Test/TestUtilityTrait.php [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+
+/**
+ * Mock class helper.
+ *
+ * @package PhpFlo\Common\Test
+ * @author Marc Aschmann <maschmann@gmail.com>
+ */
+trait TestUtilityTrait
+{
+    /**
```
Comment (maschmann) [inline src/PhpFlo/Common/Test/TestUtilityTrait.php]: Not sure if needed: If someone wants to write tests for components he should look into the component repo and there for the tests

File: src/PhpFlo/Core/Tests/Interaction/PortRegistryTest.php [hunk 5]
```diff
@@ -8,13 +8,14 @@
...
 
 use PhpFlo\Common\PortInterface;
-use PhpFlo\Interaction\ArrayPort;
-use PhpFlo\Interaction\PortRegistry;
+use PhpFlo\Core\Interaction\ArrayPort;
+use PhpFlo\Core\Interaction\PortRegistry;
+use PhpFlo\Core\Test\TestCase;
 
-class PortRegistryTest extends \PHPUnit_Framework_TestCase
+class PortRegistryTest extends TestCase
```
Comment (maschmann) [inline src/PhpFlo/Core/Tests/Interaction/PortRegistryTest.php]: The thought behind that is to have a single class with the direct phpunit dependency for the v6.x refactoring later on, where phpunit switches to namespacing.

File: src/PhpFlo/Fbp/Tests/FbpDefinitionTest.php (hunk 1 above)
Comment (maschmann) [inline src/PhpFlo/Fbp/Tests/FbpDefinitionTest.php]: maybe too big of a changeset?

**Review by aretecode — APPROVED** [review]
🛡⛑👍

**Review by aretecode — APPROVED** [review]
🔎⚒🐛🚑

**Review by aretecode — APPROVED** [review]
📜

File: bin/push-component [hunk 6]
```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
```
Comment (aretecode) [inline bin/push-component]: would be nice in a Makefile running this

**Review by aretecode — APPROVED** [review]
🎉

**Pull request merged**